### PR TITLE
Logs: Fix ui getting stuck when removing fields

### LINKED ISSUE #72504: Logs: Clicking hide this field after filtering freezes the app
In Explore, clicking on interactions in following order makes app frozen:
1. Click `show this field instead of log line` in log details
2. Click `filter for` or `filter out` some field
3. Click `hide this field`  after filtering freezes the app

https://github.com/grafana/grafana/assets/30407135/d5fc9950-3f12-4f05-80cb-39102d5c3d60

### PR DESCRIPTION
**Pull request by ivanahuckova**
How to reproduce is mentioned and showed also here: https://github.com/grafana/grafana/issues/72504: 
1. Click show this field instead of log line in log details
2. Click filter for or filter out some field
3. Click hide this field after filtering freezes the app

The issue is with `AsyncIconButton` that was introduced in and in https://github.com/grafana/grafana/pull/70328 and in  some specific cases (like the one described)  it causes  infinite loop . We should properly address the issue with `AsyncIconButton` in separate PR, but for the purposes of 10.1 release and to make fix as small as possible, we just won't use it unless experimental/WIP feature toggle `toggleLabelsInLogsUI` is enabled  and use `IconButton`.  

Fixes https://github.com/grafana/grafana/issues/72504

### CODE REVIEW

File: public/app/features/logs/components/LogDetailsRow.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { css, cx } from '@emotion/css';
 import { isEqual } from 'lodash';
 import memoizeOne from 'memoize-one';
-import React, { PureComponent, useState } from 'react';
+import React, { PureComponent, useEffect, useState } from 'react';
```
Comment (matyax) [inline public/app/features/logs/components/LogDetailsRow.tsx:4]: The use effect is not needed, right?

**Review by matyax — APPROVED** [review]
Good catch and good fix!

File: public/app/features/logs/components/LogDetailsRow.tsx (hunk 1 above)
Comment (ivanahuckova) [inline public/app/features/logs/components/LogDetailsRow.tsx:4]:
```suggestion
import React, { PureComponent, useState } from 'react';
```

Comment (ivanahuckova) [reply public/app/features/logs/components/LogDetailsRow.tsx]: yes, good catch. removed

**Pull request merged**